### ISSUE TITLE
Push deploys as an event in Datadog

### ISSUE THREAD

**Issue by jyee** [apiaryio/heroku-datadog-drain-golang#25]
It'd be really nice if this could also create an event in Datadog for code deploys. I think this would mean something like `if headers[3] == "Deploy"` around [after line 90 of logproc.go](https://github.com/apiaryio/heroku-datadog-drain-golang/blob/master/logproc.go#L90), then some corresponding new func `sendDeployEvt()` in client.go which would eventually call `c.NewEvent()` from the [Datadog library](https://godoc.org/github.com/DataDog/datadog-go/statsd#Event).

p.s. - sorry, I wish my Go skills were at a level where I could hammer this out and quickly make a PR.

**Comment by w-vi** [comment]
I am not sure if this is needed as we've got the deployment events in the datadog stream, would it make sense @abtris? Or how are we achieving the same result?

**Comment by abtris** [comment]
We use post deploy hooks for this. But this make sense. But I don't see this in heroku logs. Is some heroku docs about this?

**Comment by abtris** [comment]
There are a few options how to deploy app and we need know that works for all options (git push, pipelines, api).

**Comment by abtris** [comment]
I went thru the logs and looks as you write:

```
 Oct 07 20:29:27 <APP_NAME> heroku/api:  Release v111 created by user@company.com
```

**Comment by w-vi** [comment]
I have found in logs message like  
`Oct 07 22:29:21 my-app heroku/api: Deploy 7834b903 by user@company`

So it seems there are maybe more options depending on what way the app is deployed.

**Comment by w-vi** [comment]
Going through couple of our logs I haven't found a single line `Oct 07 20:29:27 <APP_NAME> heroku/api:  Release v111 created by user@company.com` so it is either outdated or it issued under some special conditions.

**Comment by w-vi** [comment]
I have found more things which would make sense to send as an event, basically all the activity events like rollback, setting config var etc make sense so I think we should push all of those as events. What do you think @abtris

**Comment by abtris** [comment]
I see this events in `heroku/api`

```
Oct 09 18:26:54 <APP_NAME> heroku/api:  Scale to web=6 by user@company.com
Oct 10 09:03:59 <APP_NAME> heroku/api:  Deploy 1b1facd by user@company.com
Oct 10 09:03:59 <APP_NAME> heroku/api:  Release v123 created by user@company.com
Oct 04 05:10:41 <APP_NAME> heroku/api:  Set ENV_VAR_NAME config vars by user@company.com
Sep 27 08:44:05 <APP_NAME> heroku/api:  Rollback to v111 by user@company.com
```

We can support all this changes.

**Comment by abtris** [comment]
We added support for heroku/api events using [SimpleEvent](https://godoc.org/github.com/DataDog/datadog-go/statsd#Client.SimpleEvent).

**Comment by bronislav** [comment]
I can't find api event in my event stream on data dog. Any suggestions?

**Comment by bronislav** [comment]
@abtris I've found small issue with parsing api events and tried to address it in #44

**Comment by w-vi** [comment]
Solved by #48
